packages/frontend/src/ee/features/embed/EmbedExplore/components/EmbedExplore.tsx

```diff
@@ -0,0 +1,137 @@
+import { ChartType, type SavedChart } from '@lightdash/common';
+import { MantineProvider, type MantineThemeOverride } from '@mantine/core';
+import { IconUnlink } from '@tabler/icons-react';
+import { type FC } from 'react';
+import Page from '../../../../../components/common/Page/Page';
+import SuboptimalState from '../../../../../components/common/SuboptimalState/SuboptimalState';
+import Explorer from '../../../../../components/Explorer';
+import ExploreSideBar from '../../../../../components/Explorer/ExploreSideBar';
+import { useExplore } from '../../../../../hooks/useExplore';
+import ExplorerProvider from '../../../../../providers/Explorer/ExplorerProvider';
+import { ExplorerSection } from '../../../../../providers/Explorer/types';
+import useEmbed from '../../../../providers/Embed/useEmbed';
+
+const themeOverride: MantineThemeOverride = {
+    globalStyles: () => ({
+        'html, body': {
+            backgroundColor: 'white',
+        },
+    }),
+};
+
+type Props = {
+    containerStyles?: React.CSSProperties;
+    exploreId: string;
+    savedChart: SavedChart;
+};
+
+const EmbedExplore: FC<Props> = ({
+    containerStyles,
+    exploreId,
+    savedChart,
+}) => {
+    const { projectUuid } = useEmbed();
+    const { data, error: exploreError } = useExplore(exploreId);
+
+    if (!projectUuid) {
+        return (
+            <div style={{ marginTop: '20px' }}>
+                <SuboptimalState title="Missing project UUID" />
+            </div>
+        );
+    }
+
+    if (exploreError) {
+        return (
+            <div style={{ marginTop: '20px' }}>
+                <SuboptimalState
+                    title="Error loading explore"
+                    icon={IconUnlink}
+                    description={
+                        exploreError.error.message.includes('jwt expired')
+                            ? 'This embed link has expired'
+                            : exploreError.error.message
+                    }
+                />
+            </div>
+        );
+    }
+
+    return (
+        <div style={containerStyles ?? { height: '100vh', overflowY: 'auto' }}>
+            <ExplorerProvider
+                isEditMode={true}
+                projectUuid={projectUuid}
+                savedChart={savedChart}
+                initialState={{
```

Comment:
Could this be extracted into a `getInitialState` function?

Reply:
Totally